Exercise_1.java

```diff
@@ -1,8 +1,26 @@
+
 class BinarySearch { 
     // Returns index of x if it is present in arr[l.. r], else return -1 
     int binarySearch(int arr[], int l, int r, int x) 
     { 
         //Write your code here
+        while(l<r)

```

Comment:
The terminating condition should be "while(l <= r)". For a given array [1] the program will not be able to find 1, since L and R will both be 0 and will never enter the while loop.